ed25519.go

```diff
@@ -37,7 +37,12 @@ func SupportsEd25519() bool {
 	onceSupportsEd25519.Do(func() {
 		switch vMajor {
 		case 1:
-			supportsEd25519 = versionAtOrAbove(1, 1, 1)
+			ctx := C.go_openssl_EVP_PKEY_CTX_new_id(C.GO_EVP_PKEY_ED25519, nil)
```

Comment:
No need to do this call if `versionAtOrAbove(1, 1, 1)` is false.

Reply:
```go
  if versionAtOrAbove(1, 1, 1) {
    ctx := C.go_openssl_EVP_PKEY_CTX_new_id(C.GO_EVP_PKEY_ED25519, nil)
    if ctx != nil {
     defer C.go_openssl_EVP_PKEY_CTX_free(ctx)
    }

    supportsEd25519 = ctx != nil
   }
```
IMO it is not looking any better than

```go
   ctx := C.go_openssl_EVP_PKEY_CTX_new_id(C.GO_EVP_PKEY_ED25519, nil)
   if ctx != nil {
    defer C.go_openssl_EVP_PKEY_CTX_free(ctx)
   }

   supportsEd25519 = versionAtOrAbove(1, 1, 1) && ctx != nil
   ```

Reply:
But we if we short-circuit we won't be unnecessary calling `go_openssl_EVP_PKEY_CTX_new_id`.

Reply:
since its wrapped with sync.Once, it doesn't bring any detectable costs anyway, probably few nanosecond at the very beginning of the app. But if you think its important , I'm adding that change than.

Reply:
done